cbt_export_changes.py

```diff
@@ -1,23 +1,94 @@
 #!/usr/bin/env python
 
-import sys
+"""
+For a given VDI and previous snapshot uuid this script will take a new snapshot, find the changed blocks since the previous snapshot and then export the changed blocks via nbd. This script will need to be run everytime you wish to take a new snapshot. After the changed blocks have been exported the snapshot data is destroyed to save space on the host.
+
+This script utilises the core of the new CBT functionality which is to find changes blocks between snapshots and exporting them via NBD.
+
+example: python cbt_export_changes.py -h <host address> -u <host username> -p <host password> -v <vdi uuid> -s <previous snapshot uuid> -co <changed block output path> -bo <bitmap output path>
+
+Script will then print out the new snapshot uuid at the end. Changed blocks and bitmap are saved to the paths specified.
+"""
+
 import XenAPI
+from python2_nbd_client import new_nbd_client
+import base64
+from bitstring import BitStream
+import argparse
+
+# CBT tracks 64KB blocks. Therefore each bit in the bitmap corresponds to a 64KB block on the VDI.

```

Comment:
Didn't ask in the last review, is there an API that we can get this from dynamically instead?

Reply:
There is not currently an API. One will be added if/when the block size is made variable